src/Vault.sol

```diff
@@ -309,67 +313,12 @@ contract Vault is DefaultAccessControl {
             revert NFTLimitExceeded();
         }
 
-        _requireVaultOwner(vaultId);
-
-        {
-            (
-                ,
-                ,
-                address token0,
-                address token1,
-                uint24 fee,
-                int24 tickLower,
-                int24 tickUpper,
-                uint128 liquidity,
-                uint256 feeGrowthInside0LastX128,
-                uint256 feeGrowthInside1LastX128,
-                uint128 tokensOwed0,
-                uint128 tokensOwed1
-            ) = positionManager.positions(nft);
-
-            positionManager.transferFrom(msg.sender, address(this), nft);
-            {
-                uint160 sqrtRatioAX96 = TickMath.getSqrtRatioAtTick(tickLower);
-                uint160 sqrtRatioBX96 = TickMath.getSqrtRatioAtTick(tickUpper);
-
-                _uniV3PositionInfo[nft] = UniV3PositionInfo({
-                    token0: token0,
-                    token1: token1,
-                    targetPool: IUniswapV3Pool(factory.getPool(token0, token1, fee)),
-                    vaultId: vaultId,
-                    sqrtRatioAX96: sqrtRatioAX96,
-                    sqrtRatioBX96: sqrtRatioBX96
-                });
-
-                if (!protocolGovernance.isPoolWhitelisted(factory.getPool(token0, token1, fee))) {
-                    revert InvalidPool();
-                }
-
-                if (!oracle.hasOracle(token0) || !oracle.hasOracle(token1)) {
-                    revert MissingOracle();
-                }
-            }
-
-            if (
-                _calculateAdjustedCollateral(
-                    _uniV3PositionInfo[nft],
-                    DENOMINATOR,
-                    UniswapV3FeesCalculation.PositionInfo({
-                        tickLower: tickLower,
-                        tickUpper: tickUpper,
-                        liquidity: liquidity,
-                        feeGrowthInside0LastX128: feeGrowthInside0LastX128,
-                        feeGrowthInside1LastX128: feeGrowthInside1LastX128,
-                        tokensOwed0: tokensOwed0,
-                        tokensOwed1: tokensOwed1
-                    })
-                ) < protocolGovernance.protocolParams().minSingleNftCollateral
-            ) {
-                revert CollateralUnderflow();
-            }
+        if (vaultOwner[vaultId] == address(0)) {
+            revert InvalidVault();
         }
 
-        _vaultNfts[vaultId].add(nft);
+        positionManager.transferFrom(msg.sender, address(this), nft);
```

Comment:
I think we can use `safeTransferFrom` here as well, so that collateral would be deposited in `onERC721Received`.